backend/controllers/auth.js

```diff
@@ -23,6 +25,28 @@ module.exports = {
       );
       return;
     }
+    res.send(ResponseHelper.generateResponse(200, "Success", { user }));
+  },
+  signup: async (req, res) => {
+    const { first_name, last_name, email, password } = req.body;
+    const existingUser = await User.findOne({ where: { email } });
+
+    if (existingUser) {
+      res.send(
+        ResponseHelper.generateResponse(401, "Email is already in use.")
+      );
+      return;
+    }
+
+    const hash = await bcrypt.hash(password, salt);
+    const user = await User.create({
+      first_name,
+      last_name,
+      email,
+      password: hash,
+      user_type: "user",
+      avatar_url: "default_img.jpeg",
+    });
 
     res.send(ResponseHelper.generateResponse(200, "Success", { user }));
```

Comment:
- you don't need to return the entire user data, especially the password. The id and email and timestamps would be good na.